networkit/cpp/io/NetworkitBinaryReader.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * NetworkitBinaryReader.cpp
+ *
+ *@author Charmaine Ndolo <charmaine.ndolo@b-tu.de>
+ */
+
+#include <networkit/io/NetworkitBinaryReader.hpp>
+#include <networkit/io/MemoryMappedFile.hpp>
+#include <networkit/graph/GraphBuilder.hpp>
+
+namespace NetworKit {
+
+size_t NetworkitBinaryReader::decode(const uint8_t* data, uint64_t& value) {
+	int n;
+	if(!data[0]) {
+		n = 8;
+		value = 0;
+	} else {
+		n = __builtin_ctz(data[0]);
```

Comment:
Here, we can use `ffs` from TLX.

Reply:
However, `ffs` (in https://github.com/tlx/tlx/blob/master/tlx/math/ffs.hpp), returns the number of trailing zeros + 1, so we have to take `ffs(data[0]) - 1`.